diff --git a/visualization/CMakeLists.txt b/visualization/CMakeLists.txt
index 6e6ce0c7d..cd41e1f1f 100644
--- a/visualization/CMakeLists.txt
+++ b/visualization/CMakeLists.txt
@@ -143,6 +143,7 @@ target_link_libraries("${LIB_NAME}" pcl_common pcl_io pcl_kdtree pcl_geometry pc
 if(${VTK_VERSION} VERSION_GREATER_EQUAL 9.0)
   #Some libs are referenced through depending on IO
   target_link_libraries("${LIB_NAME}"
+                        X11
                         VTK::ChartsCore
                         VTK::CommonColor
                         VTK::CommonComputationalGeometry
