--- a/build_scripts/platforms/unix.py
+++ b/build_scripts/platforms/unix.py
@@ -108,10 +108,10 @@ def prepare_packages_posix(pyside_build, _vars, cross_build=False):
             "{st_build_dir}/{st_package_name}/scripts/shiboken_tool.py",
             force=False, _vars=_vars)
 
-    if config.is_internal_shiboken_generator_build() or config.is_internal_pyside_build():
+    if config.is_internal_shiboken_module_build() or config.is_internal_pyside_build():
         # <install>/include/* -> <setup>/{st_package_name}/include
         copydir(
-            "{install_dir}/{cmake_package_name}/include",
+            "{install_dir}/include/{cmake_package_name}",
             destination_dir / "include",
             _vars=_vars)

@@ -461,7 +467,7 @@
 		popd >/dev/null || die
 	fi
 	mkdir -p "${BUILD_DIR}/install/usr/include"
-	for dir in PySide6 shiboken6_generator; do
+	for dir in PySide6 shiboken6; do
 		if [[ -d ${BUILD_DIR}/install/$(python_get_sitedir)/${dir}/include ]]
 		then
 			ln -s "../../$(python_get_sitedir)/${dir}/include" \
